staging/provisioning/windows/kubeletstart.ps1

```diff
@@ -247,12 +246,15 @@ if ($global:NetworkPlugin -eq "kubenet") {
             # Add route to all other POD networks
             Update-CNIConfigKubenetDocker $podCIDR $masterSubnetGW
         }
-
-        # startup the service
-        Invoke-Expression $KubeletCommandLine
     }
     catch {
         Write-Error $_
     }
-
 }
+
+# Start the kubelet
+# Use run-process.cs to set process priority class as 'High'
+Add-Type -Path .\run-process.cs
+$exe = "$global:KubeDir\kubelet.exe"
+$args = ($KubeletArgList -join " ")
+[RunProcess.exec]::RunProcess($exe, $args)
```

Comment:
Will it also set a high priority for all child processes created by kubelet, for example, cni plugin?

Reply:
Great question.
I'll try setting the (BasePriority)[https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.basepriority?view=net-5.0] for the process as well.
The docs say that controls the starting priority of new threads in the process. 
I don't know the details of how go go starts new processes for Windows or this will have an effect but it might be worth testingout.

Reply:
It does not look like there is a way for Windows to set the priority of new processes.

Reply:
Thanks for the info. I am curious what will happen if there is no enough cpu resource for CNI plugin. I think that it should be ok if the IP cannot be allocated so the new container cannot be created. But can the container be terminated successfully if the IP cannot be released successfully?

Reply:
I'm not exactly sure what would happen.

@kevpar proposed a solution where a process can create an NT job object, set a priority class to that job object, then add itself to that job object. That solution should cause processes spawned by kubelet/kube-proxy/containerd to run with higher priority.

I'd like to try and get these changes merged and iterate if needed.
I think a better solution would be to have each process implement the above behavior and then we can remove the run-process work at a future time.